### PR TITLE
Support lr_staleness_modulation in python PS

### PR DESCRIPTION
**Pull request by skydoorkai**
This PR only works with learning_rate_scheduler so workers can send learning_rate together with gradients.

Modulating learning_rate defined in the optimizer is not implemented in this PR.

### CODE REVIEW

**Review by QiJune — APPROVED** [review]
Excellent!
